ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -232,6 +241,24 @@ public int persistColumnStats(Hive db, Table tbl) throws HiveException, MetaExce
   public void setDpPartSpecs(Collection<Partition> dpPartSpecs) {
   }
 
+  private void setOrRemoveColumnStatsAccurateProperty(Hive db, TableName tableName, List<String> colNames, boolean success) throws HiveException {
+    if (CollectionUtils.isEmpty(colNames) || !colStatDesc.isTblLevel()) {
+      return;
+    }
+    EnvironmentContext environmentContext = new EnvironmentContext();
+    environmentContext.putToProperties(StatsSetupConst.DO_NOT_UPDATE_STATS, StatsSetupConst.TRUE);
+    // Fetch latest table object
+    Table newTable = db.getTable(tableName.getDbTable());
+    // Lowercase all column names.
+    colNames = colNames.stream().map(x -> x.toLowerCase(Locale.ROOT)).collect(Collectors.toList());
```

Comment:
you are modifying the globally declared object here

Reply:
Modified the array to lowercase without modifying the source. This is done in ColumnStatsDesc. Done.